tests/aws/services/stepfunctions/v2/error_handling/test_states_errors.py

```diff
@@ -0,0 +1,176 @@
+import json
+
+from localstack_snapshot.snapshots.transformer import RegexTransformer
+
+from localstack.testing.pytest import markers
+from localstack.utils.strings import short_uid
+from tests.aws.services.stepfunctions.templates.errorhandling.error_handling_templates import (
+    ErrorHandlingTemplate as EHT,
+)
+from tests.aws.services.stepfunctions.utils import create_and_record_execution
+
+
+@markers.snapshot.skip_snapshot_verify(
+    paths=[
+        "$..loggingConfiguration",
+        "$..tracingConfiguration",
+    ]
+)
+class TestStatesErrors:
```

Comment:
Thorough coverage of error scenarios 👏 

💡  A short high-level overview of the key test cases (e.g., as class docstring) could improve clarity. That would clarify the two cases `service_task_lambda` and `task_lambda`.

Reply:
More of a thought input, rather than actionable:
🧠  Given the inherent complexity of StepFunctions, it's impossible to test all possible scenarios. I wonder whether you have some intuition for a testing strategy to provide good enough coverage while keeping an eye on the test execution time?